AreaCalculator.java

```diff
@@ -1,64 +1,81 @@
 
 public class AreaCalculator {
 	
-	  public static double calculate(String... params) { 
+	  public static double calculate(String... params) throws NotEnoughParamException,NumberFormatException, NegativeLengthException { 
```

Comment:
When using exceptions, you need to think: will it annoy my users (the API users, I mean), or will it help them understand what kind of errors could happen when using my method?
Also, you could ask yourself - is there a real difference between the exceptions, that my user could and should behave differently when catching them, or are they similar?
See [RuntimeException](https://docs.oracle.com/javase/10/docs/api/java/lang/RuntimeException.html), and think about it. It might not be the right solution, but I think you should be able to explain it, either way. :)

Reply:
The 3 types of exceptions, I mentioned in the code, should extend from RuntimeException. Because they appear when the user has a problem with his inputs.
For each exception, the user should act differently.